runtime/vm/createramclass.cpp

```diff
@@ -1627,6 +1631,60 @@ loadSuperClassAndInterfaces(J9VMThread *vmThread, J9ClassLoader *classLoader, J9
 	return TRUE;
 }
 
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+/**
+ * Attempts to pre-load classes of fields with Q signatures and no static
+ * access modifier set.
+ *
+ * Caller should not hold the classTableMutex.
+ *
+ * Return TRUE on success. On failure, returns FALSE and sets the
+ * appropriate Java error on the VM.
+ */
+static VMINLINE BOOLEAN
+loadFlattenableFieldValueClasses(J9VMThread *vmThread, J9JavaVM *vm, J9ClassLoader *classLoader, J9ROMClass *romClass, UDATA classPreloadFlags, UDATA packageID, J9Module *module)
```

Comment:
As a rule, don't pass thread and VM (even though you have both in hand in the caller. Not sure this needs to be force inlined. Please use "currentThread" rather than "vmThread" for new code.